nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/conf/bootstrap.conf

```diff
@@ -90,6 +85,13 @@ java.arg.17=-Dzookeeper.admin.enableServer=false
 #            org.apache.jasper.servlet.JasperLoader,org.jvnet.hk2.internal.DelegatingClassLoader,org.apache.nifi.nar.NarClassLoader
 # End of Java Agent config for native library loading.
 
+# The following entry is needed in Java 21 because some libraries invoke
+# reflective calls that Java no longer considers allowed by default.
+# https://docs.oracle.com/en/java/javase/16/migrate/migrating-jdk-8-later-jdk-releases.html#GUID-7BB28E4D-99B3-4078-BDC4-FC24180CE82B
+# This may need to be modified if additional reflective access is needed by certain libraries
+# This is only known to be needed for the Hive3 processors as of now.
+java.arg.20=--add-opens=java.base/java.net=ALL-UNNAMED
```

Comment:
Although this is an unfortunate requirement for Hive 3, the comment is helpful. Hopefully we can remove this in the future with newer Hive versions.